src/foam/nanos/auth/CountryAndRegionCacheService.java

```diff
@@ -0,0 +1,163 @@
+package foam.nanos.auth;
+
+import foam.core.ContextAwareSupport;
+import foam.dao.ArraySink;
+import foam.dao.DAO;
+import foam.dao.Sink;
+import foam.mlang.MLang;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.NanoService;
+import foam.nanos.notification.email.EmailMessage;
+import foam.nanos.notification.email.EmailService;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * Query the country and region by code, name, or alternative names.
+ */
+public class CountryAndRegionCacheService extends ContextAwareSupport implements NanoService {
```

Comment:
Can you split these into two classes.

Reply:
Yes, seems to be two entirely different things: email notifications and cachine.

Also, why is caching needed? Data is already stored in memory, so how does this help?

Reply:
@kgrgreer I was talking about the split of Country and Region.

Reply:
re: caching. we can use IN to test the alternates. 
IN(country.getAlternatives())
This can replace all the caching code.

Reply:
I wasn't aware of the MLang.IN before.